idl/github.com/uber/cadence/shared.thrift

```diff
@@ -825,6 +831,10 @@ struct DomainInfo {
 struct DomainConfiguration {
   10: optional i32 workflowExecutionRetentionPeriodInDays
   20: optional bool emitMetric
+  30: optional string archivalBucketName
+  40: optional i32 archivalRetentionPeriodInDays
+  50: optional ArchivalStatus archivalStatus
+  60: optional string archivalBucketOwner
```

Comment:
why do we need to have a separate owner information for the bucket as part of domain configuration?  This is more like bucket metadata and should belong to blob store.